pages/index.js

```diff
@@ -1,129 +1,95 @@
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faPlus } from "@fortawesome/free-solid-svg-icons";
 import AddCalories from "../components/AddCalories/AddCalories";
 import useCalorieStore from "../utils/useCalorieStore";
+import { unixDate } from "../utils/useCalorieStore";
 import ConsumedList from "../components/ConsumedList/ConsumedList";
 import HomeCalendar from "../components/Calendar";
-import useLocalStorageState from "use-local-storage-state";
 import { useState, useEffect } from "react";
 import {
   StyledDiv,
   StyledButtonCalorieCounter,
   ConsumedContainer,
   FormContainer,
   OpenCalorieFormButton,
+  LoadingDisplay,
 } from "../components/IndexPage/styles";
 
 export default function HomePage() {
-  const {
-    dailyCount,
-    dailyMeals,
-    resetDailyMeals,
-    calorieGoal,
-    setDailyCount,
-    addHistoryEntry,
-    history,
-  } = useCalorieStore();
+  const { history, calorieGoals } = useCalorieStore();
 
   const [isLoading, setIsLoading] = useState(true);
-
   const [isListVisible, setIsListVisible] = useState(false);
-  const [isGoalExceeded, setIsGoalExceeded] = useState(false);
   const [isFormVisible, setIsFormVisible] = useState(false);
 
-  // reset the daily calorie count & save history entry
-  const [currentDay, setCurrentDay] = useLocalStorageState("currentDay", {
-    defaultValue: new Date().getDate(),
-  });
-
-  setInterval(() => {
-    const today = new Date();
-    if (today.getDate() !== currentDay) {
-      addHistoryEntry(
-        new Date().getFullYear(),
-        new Date().getMonth(),
-        new Date().getDate(),
-        isGoalExceeded
-      );
-      setCurrentDay(today);
-      setDailyCount(0);
-      resetDailyMeals();
-    }
-  }, 5000);
-  // <----------------------------------------
-
   function handleWindowClosing() {
     setIsListVisible(false);
     setIsFormVisible(false);
   }
 
+  useEffect(() => {
+    setIsLoading(false);
+  }, []);
+
+  useEffect(() => {
+    if (!history.some((entry) => entry.date === unixDate)) {
+      setIsListVisible(false);
+    }
+  }, [history]);
+
   useEffect(() => {
```

Comment:
couldn't you use a normal `hadlerFunction` to get rid of the useEffect?

Reply:
Thank you for pointing that out. However, I can't find a solution for the list to render as intended without the useEffect :(